### PR TITLE
[BugFix] when forward query to leader fe forward the current catalog in connect context

### PR DESCRIPTION
**Pull request by zombee0**
Signed-off-by: zombee0 <flylucas_10@163.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/StarRocksTest/issues/1831

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -548,6 +548,7 @@ private void finalizeCommand() throws IOException {
     }
 
     public TMasterOpResult proxyExecute(TMasterOpRequest request) {
+        ctx.setCurrentCatalog(request.catalog);
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:551]: If old version forward to new version, the request.catalog will be null. Will it report NEP?

Comment (zombee0) [reply fe/fe-core/src/main/java/com/starrocks/qe/ConnectProcessor.java:551]: There will no be NPE, but we shall give back a message to tell user upgrade

**Review by gengjun-git — APPROVED** [review]

**Comment by zombee0** [comment]
run starrocks_fe_unittest

**Review by stephen-shelby — APPROVED** [review]

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 8 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/LeaderOpExecutor.java|0|1|00.00%|[137]|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|0|7|00.00%|[551, 552, 556, 557, 559, 560, 561]|

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by yongbingwang** [comment]
branch-2.5
https://github.com/StarRocks/starrocks/pull/17379

**Comment by yongbingwang** [comment]
branch-2.3
https://github.com/StarRocks/starrocks/pull/17377

**Pull request merged**